### ISSUE TITLE
Fatal plant loop temperature error appears to originate in some water to air heat pumps in the very first warmup timestep

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#5936]
Issue overview
--------------
User file which ran to completion in v8.5, fails in v8.6 with a fatal error right at the first timestep during warmup:
```
   ** Severe  ** Plant temperatures are getting far too hot, check controls and relative loads and capacities
   **   ~~~   **  During Warmup, Environment=RUN PERIOD 1, at Simulation time=01/01 00:00 - 00:15
   **   ~~~   ** PlantLoop Name (DemandSide)= CONDENSER WATER LOOP
   **   ~~~   ** PlantLoop Setpoint Temperature=7.3 {C}
   **   ~~~   ** PlantLoop Inlet Node (SupplySide) does not have a Setpoint.
   **   ~~~   ** PlantLoop Inlet Node (DemandSide) does not have a Setpoint.
   **   ~~~   ** PlantLoop Outlet Node (SupplySide) does not have a Setpoint.
   **   ~~~   ** PlantLoop Outlet Node (DemandSide) does not have a Setpoint.
   **   ~~~   ** PlantLoop Outlet Node (DemandSide) "NODE 412" has temperature=3168.3 {C}
```

Running this in the debugger shows that these two nodes are also >3000C, so these heat pumps are likely the source of the problem:
```
  Coil:Cooling:WaterToAirHeatPump:EquationFit,
    Flr1EmergencyPower Water to Air Heat Pump Cooling Coil,  !- Name
    Node 1065,               !- Water Outlet Node Name

  Coil:Cooling:WaterToAirHeatPump:EquationFit,
    Flr1MPPO Water to Air Heat Pump Cooling Coil,  !- Name
    Node 1089,               !- Water Outlet Node Name
```

### Workaround
Set the "Maximum Loop Temperature" to say 3000C for "Condenser Water Loop", then the fatal error is not tripped and the situation corrects itself (probably in the next few iterations) and the simulation completes.  

### Details
Some additional details for this issue (if relevant):
 - Platform Win64
 - Version of EnergyPlus v8.6
 - Helpdesk ticket number11690

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added to EnergyPlusDevSupport\DefectFiles, use USA_AZ_Phoenix-Sky.Harbor.Intl.AP.722780_TMY3.epw
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by KWSL** [comment]
I understand that changes were made to the variable water flow algorithm for water to air heat pumps for version 8.6. As the person who originally wrote the variable water flow algorithm, I remember having similar issues with my original code. The current variable flow algorithm is probably allowing very, very low water flow rates, which in turn result in high temperatures. 

What is the maximum temperature on the loop over a year-long simulation? If it only shows up in sizing, does it make sense to run the sizing party of the simulation with the heat pumps in constant flow mode?

**Comment by rraustad** [comment]
The model expects to see design water flow rate and requests that value in Init. The plant chokes the flow from 0.9 kg/s to 0.001 kg/s and causes a very high outlet water temp.

    SimpleWatertoAirHP( HPNum ).WaterMassFlowRate = SimpleWatertoAirHP( HPNum ).DesignWaterMassFlowRate;
    SetComponentFlowRate( SimpleWatertoAirHP( HPNum ).WaterMassFlowRate, ....;

So the model expects to be using design values when calculating outlet water temp, but what if the plant changes the water flow rate? If within reason the temps should be tolerable, but if extreme difference from design the outlet temp can be extreme. The defect file has mixed hard values and autosized for the demand side yet hard size for the pump and tower. UPDATE: No, not mixed sizing, all WSHP cooling coil water flow are hard sized. There is one WSHP heating coil that is autosized.

    Coil:Heating:WaterToAirHeatPump:EquationFit,
      Flr1Security Water to Air Heat Pump Heating Coil,  !- Name
      0.8967001417,            !- Rated Air Flow Rate {m3/s}
      Autosize,                !- Rated Water Flow Rate {m3/s}
      Autosize,                !- Gross Rated Heating Capacity {W}

    Pump:VariableSpeed,
      Condenser Water Loop Pump,  !- Name
      0.112288670199344,       !- Design Maximum Flow Rate {m3/s}

    CoolingTower:VariableSpeed,
      0.112288670199344,       !- Design Water Flow Rate {m3/s}


    SimpleWatertoAirHP( HPNum ).OutletWaterTemp = SourceSideInletTemp + QSource_fullload / ( SourceSideMassFlowRate * CpWater );

Not sure of the fix yet.

One thought is that the model coefficients should account for this (ratioVS = 0.0014 here):

    if ( SimpleWatertoAirHP( HPNum ).DesignWaterMassFlowRate > 0.0 ) {
      ratioVS = ( SourceSideMassFlowRate ) / ( SimpleWatertoAirHP( HPNum ).DesignWaterMassFlowRate );
    } else {
      ratioVS = 0.0;
    }

    QLoadTotal = TotalCapRated * ( TotalCapCoeff1 + ( ratioTWB * TotalCapCoeff2 ) + ( ratioTS * TotalCapCoeff3 ) + ( ratioVL * TotalCapCoeff4 ) + ( ratioVS * TotalCapCoeff5 ) );

Another thought is that the model should shut down in certain cases but the user would need to know why.

	//Check for flows, do not perform simulation if no flow in load side or source side.
	if ( SourceSideMassFlowRate <= 0.0 || LoadSideMassFlowRate <= 0.0 ) {
		SimpleWatertoAirHP( HPNum ).SimFlag = false;
		return;

**Comment by mjwitte** [comment]
Is this related to #5919?

**Comment by rraustad** [comment]
This defect file runs with the correction to #5919. See pull request #6035.

**Comment by Myoldmopar** [comment]
Closed via #6035
